core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -251,6 +251,13 @@ private[spark] object JettyUtils extends Logging {
   private def attachPrefix(basePath: String, relativePath: String): String = {
     if (basePath == "") relativePath else (basePath + relativePath).stripSuffix("/")
   }
+
+  /**
+   * Attach a prefix to the given path, don't remove the trailing / if present.
+   */
+  private def attachPrefixForRedirect(basePath: String, relativePath: String): String = {
```

Comment:
Both of these methods seems superfluous. They're trivial and each called once. I don't see why `basePath == ""` is treated separately, like, why the trailing slash wouldn't be stripped. In fact, there's only one way a call sets `basePath` to anything but `""` and it's the `/jobs` servlet below - not clear why only that is special cased?

Anyway I'd suggest removing the methods and removing the special-casing -- it's just the combination of two strings, with a trailing / stripped in one case.